pandas/tests/io/formats/test_format.py

```diff
@@ -3432,3 +3432,14 @@ def test_format_remove_leading_space_dataframe(input_array, expected):
     # GH: 24980
     df = pd.DataFrame(input_array).to_string(index=False)
     assert df == expected
+
+
+@pytest.mark.parametrize("na_rep, string", [("NaN", "nan"), ("Ted", "Ted")])
```

Comment:
i'm not sure about using "nan" when `na_rep` is "NaN", although that is the string output of `"{:.2f}".format(np.nan)`.

Maybe could use `lib.no_default` for `na_rep`, raise if `float_format` specified and use "NaN" if `float_format` not specified.

maybe worth considering passing `np.nan` to `func` if passed to `float_format`, and if result is a string assume `func` handles missing values, something like

```
try:
   func_handles_na = isinstance(func(np.nan), str)
except Execption:
   func_handles_na = False
```

and then use `na_rep` if `func_handles_na` is `False`. but I'm not sure if we gain anything.

The main issue with the custom formatters (applies to `formatters` kwarg as well) is that we do not give complete control to the custom formatter. off the top of my head, strings maybe trimmed, spaces added, precision changed, truncation applied.

The other issue is that the EAs use the custom formatters. So this is not necessarily an easy issue to fix in isolation.